test/functional/feature_evm_state_root_change.py

```diff
@@ -0,0 +1,37 @@
+from test_framework.test_framework import DefiTestFramework
+from test_framework.util import (
+    assert_equal
+)
+
+class StateRootChangeTest(DefiTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.setup_clean_chain = True
+        self.extra_args = [
+            ['-txordering=2', '-dummypos=0', '-txnotokens=0', '-amkheight=50', '-bayfrontheight=51', '-eunosheight=80', '-fortcanningheight=82', '-fortcanninghillheight=84', '-fortcanningroadheight=86', '-fortcanningcrunchheight=88', '-fortcanningspringheight=90', '-fortcanninggreatworldheight=94', '-fortcanningepilogueheight=96', '-grandcentralheight=101', '-nextnetworkupgradeheight=105', '-changiintermediateheight=105', '-changiintermediate2height=105', '-changiintermediate3height=105', '-changiintermediate4height=110', '-subsidytest=1', '-txindex=1'],
+        ]
+
+    def run_test(self):
+        node = self.nodes[0]
+        node.generate(105)
+
+        # Activate EVM
+        self.nodes[0].setgov({"ATTRIBUTES": {'v0/params/feature/evm': 'true'}})
+        self.nodes[0].generate(1)
+
+        # check counter contract
+        from web3 import Web3
+        w3 = Web3(Web3.HTTPProvider(self.nodes[0].get_evm_rpc()))
+
+        NUM_BLOCKS = 5
+        state_roots = set()
+        for i in range(NUM_BLOCKS):
+            node.generate(1)
+            block = w3.eth.get_block('latest')
+            state_roots.add(Web3.to_hex(block["stateRoot"]))
+
+        assert_equal(len(state_roots), NUM_BLOCKS)
```

Comment:
hmm.. length can't assert the state_root is different in every block...
need `assert_not_equal` to support this test.. eg: 
```
b0 = w3.eth.get_block('latest')
node.generate(1)
b1 = w3.eth.get_block('latest')
assert_not_equal(b0['state_root'], b1['state_root'])
```

Reply:
`state_roots` is a set so it removes duplicate entries during `add`. So if any two blocks have the same hash, the set would be one element shorter than expected and the test would fail.

Running this test on a master build fails
```
2023-07-24T07:19:01.369000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/shohamc1/projects/ain/test/functional/test_framework/test_framework.py", line 217, in main
    self.run_test()
  File "/home/shohamc1/projects/ain/test/functional/feature_evm_state_root_change.py", line 33, in run_test
    assert_equal(len(state_roots), NUM_BLOCKS)
  File "/home/shohamc1/projects/ain/test/functional/test_framework/util.py", line 41, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(1 == 5)
```